proto/build.rs

```diff
@@ -40,3 +50,40 @@ fn main() -> Result<()> {
 
     Ok(())
 }
+
+// WARNING: any type attributes adding SERDE_AS **MUST** be placed **BEFORE**
+// SERIALIZE. Otherwise, serde_as is applied to the output of the derive macro
+// (unstable), rather than the other way around.
+// This is a moot point for now, since we can't use field attributes for macro reasons
+// static SERDE_AS: &str = r#"#[::serde_with::serde_as]"#;
+static SERIALIZE: &str = r#"#[derive(::serde::Deserialize, ::serde::Serialize)]"#;
+
+// Requires SERDE_AS on the container
+// :(
+// error: expected non-macro attribute, found attribute macro `::serde_with::serde_as`
+// use tendermint-rs approach instead?
+// https://github.com/penumbra-zone/tendermint-rs/blob/master/proto/src/serializers/bytes.rs#L4-L30
+// static AS_HEX: &str = r#"#[::serde_with::serde_as(as = "::serde_with::hex::Hex")]"#;
+
+static AS_HEX: &str = r#"#[serde(with = "crate::serializers::hexstr")]"#;
+static AS_BASE64: &str = r#"#[serde(with = "crate::serializers::base64str")]"#;
```

Comment:
Some of this could be deleted; initially, I tried to use `serde_with`, but ran into problems adding the attribute macros to generated code output by prost. It seems kind of brittle anyways, and the `tendermint-rs` approach works pretty well.